src/object.c

```diff
@@ -779,14 +780,17 @@ size_t stringObjectLen(robj *o) {
 
 int getDoubleFromObject(const robj *o, double *target) {
     double value;
+    char *eptr;
 
     if (o == NULL) {
         value = 0;
     } else {
         serverAssertWithInfo(NULL,o,o->type == OBJ_STRING);
         if (sdsEncodedObject(o)) {
-            if (!string2d(o->ptr, sdslen(o->ptr), &value))
-                return C_ERR;
+            value = fast_float_strtod(o->ptr, &eptr);
+            /* check end pointer is not the null terminator, that the value a number,
+               and that the pointer to the last char is not equal to the first (empty string passed) */
+            if (eptr[0] != '\0' || isnan(value) || (eptr[0] == ((char *)o->ptr)[0])) return C_ERR;
```

Comment:
why not keep the old `string2d()` which also uses fast_float_strtod()?

Reply:
indeed. Addressed in de8b1beb227919185cb53ce648ce2c3fa9d9b491